### PR TITLE
fix(thumbnail): border in high contrast mode

### PR DESCRIPTION
**Pull request by jawinn**
## Description

**fix(thumbnail): high contrast mode border**

Adds missing border to high contrast mode, for the border created with `box-shadow`. High contrast does not display box-shadows.

Also makes sure the existing high contrast styles are within the root thumbnail class selector, and not within global `.is-selected` or `is-focused` classes.

**feat(thumbnail): add some controls to storybook**

Add controls to Storybook for the three args that are already used: isSelected, isLayer, and backgroundColor.
Also re-categorizes a few controls into "State" and "Content".

## How and where has this been tested?
 - Chromed forced-colors emulation, dark and light.
 - AssistivTunnel, Edge - Windows High Contrast Mode, dark and light themes

## Screenshots
Before:
Screenshot 2023-05-18 at 10 40 43 AM
After:
Screenshot 2023-05-18 at 10 41 17 AM

## To-do list

- [x] If my change impacts other components, I have tested to make sure they don't break.
- [x] If my change impacts documentation, I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).
- [x] I have tested these changes in Windows High Contrast mode.
- [x] I have updated any relevant storybook stories and templates. 
- [ ] If my change(s) include visual change(s), a designer has reviewed and approved those changes.
- [x] This pull request is ready to merge.

### CODE REVIEW

**Review by jenndiaz — COMMENTED** [review]
Just a couple of questions!

Should we also be displaying the checkerboard in WHCM? I noticed we do with Swatch.

File: components/thumbnail/index.css [hunk 1]
```diff
@@ -99,7 +99,7 @@ governing permissions and limitations under the License.
     repeating-conic-gradient(var(--mod-thumbnail-checkerboard-light-color, var(--spectrum-thumbnail-checkerboard-light-color)) 0% 25%, var(--mod-thumbnail-checkerboard-dark-color, var(--spectrum-thumbnail-checkerboard-dark-color)) 0% 50%)
     left top / calc(var(--mod-thumbnail-checkerboard-size, var(--spectrum-thumbnail-checkerboard-size)) * 2) calc(var(--mod-thumbnail-checkerboard-size, var(--spectrum-thumbnail-checkerboard-size)) * 2);
 
-  &:before {
+  &::before {
```
[the remark this answers is not in the document]
Comment (jawinn) [reply components/thumbnail/index.css:102]: It's kind of a half dozen of one, 6 of the other situation. It technically should be `::` now unless we were supporting IE8 and earlier. Most browsers support the old `:` syntax for backwards compatibility. I'd imagine the build system could convert them if we really wanted all before and after `::` to be `:`. I was not a necessary change for the high contrast fix; just something I felt should be updating when looking through the file. I do see other components using both versions of the syntax, it's kind of just mixed between usage of both based on when they've been worked on etc.

File: components/thumbnail/stories/thumbnail.stories.js [hunk 2]
```diff
@@ -61,7 +90,9 @@ export default {
     rootClass: "spectrum-Thumbnail",
     size: "500",
     isCover: false,
+    isLayer: false,
     isDisabled: false,
+    isSelected: false,
```
[the remark this answers is not in the document]
Comment (jawinn) [reply components/thumbnail/stories/thumbnail.stories.js:95]: Let me defer to @pfulton or @castastrophe on that question. I'd think the latter; it had me questioning whether isSelected should have a visual change when isLayer is false, and I just wasn't able to test it.

**Comment by jawinn** [comment]
> Just a couple of questions!
> 
> Should we also be displaying the checkerboard in WHCM? I noticed we do with Swatch.

Great question. I don't know which way is better. Looking at the landscape image with the checkerboard behind in seems less contrasty to my eyes than with the black background. If that's wanted, we'll have to use `forced-color-adjust: none;` and make some other adjustments.

File: components/thumbnail/index.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (jenndiaz) [reply components/thumbnail/index.css:102]: ok thanks for the context, I'll stick to `::` moving forward.

File: components/thumbnail/stories/thumbnail.stories.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (pfulton) [reply components/thumbnail/stories/thumbnail.stories.js:95]: @jenndiaz @jawinn I think it makes sense to show the control, but disable it if we can: https://storybook.js.org/docs/react/essentials/controls#conditional-controls

[the remark this answers is not in the document]
Comment (jawinn) [reply components/thumbnail/stories/thumbnail.stories.js:96]: I've just added the conditional control, so isSelected only shows when isLayer is enabled.

**Review by jenndiaz — APPROVED** [review]
Looks great!

**Review by pfulton — APPROVED** [review]

**Pull request merged**